examples/ChatQnA/deploy/k8s_helm.md

```diff
@@ -0,0 +1,497 @@
+Proxy can be set in the `values.yaml`.
+Open the `values.yaml` file using an editor
+```bash
+vi chatqna/values.yaml
+```
+Update the following section and save file:
+```yaml
+# chatqna/values.yaml
+global:
+  http_proxy: "http://your-proxy-address:port"
+  https_proxy: "http://your-proxy-address:port"
+  no_proxy: "localhost,127.0.0.1,localaddress,.localdomain.com"
+```
+## Use Case Setup
+
+The `GenAIInfra` repository utilizes a structured Helm chart approach, comprising a primary `Charts.yaml` and individual sub-charts for components like the LLM Service, Embedding Service, and Reranking Service. Each sub-chart includes its own `values.yaml` file, enabling specific configurations such as Docker image sources and deployment parameters. This modular design facilitates flexible, scalable deployment and easy management of the GenAI application suite within Kubernetes environments. For detailed configurations and common components, visit the [GenAIInfra common components directory](https://github.com/opea-project/GenAIInfra/tree/main/helm-charts/common).
+
+This use case employs a tailored combination of Helm charts and `values.yaml` configurations to deploy the following components and tools:
+|use case components | Tools |   Model     | Service Type |
+|----------------     |--------------|-----------------------------|-------|
+|Data Prep            |  LangChain   | NA                       |OPEA Microservice |
+|VectorDB             |  Redis       | NA                       |Open source service|
+|Embedding            |   TEI        | BAAI/bge-base-en-v1.5    |OPEA Microservice |
+|Reranking            |   TEI        | BAAI/bge-reranker-base   | OPEA Microservice |
+|LLM                  |   TGI        |Intel/neural-chat-7b-v3-3 |OPEA Microservice |
+|UI                   |              | NA                       | Gateway Service |
+
+Tools and models mentioned in the table are configurable either through the
+environment variable or `values.yaml` 
+
+Set a new [namespace](#create-and-set-namespace) and switch to it if needed
+
+To enable UI, uncomment the lines `56-62` in `GenAIInfra/helm-charts/chatqna/values.yaml`:
+```bash
+chatqna-ui:
+   image:
+     repository: "opea/chatqna-ui"
+     tag: "latest"
+   containerPort: "5173"
+```
+
+Next, we will update the dependencies for all Helm charts in the specified directory and ensure the `chatqna` Helm chart is ready for deployment by updating its dependencies as defined in the `Chart.yaml` file.
+
+```bash
+# all Helm charts in the specified directory have their 
+# dependencies up-to-date, facilitating consistent deployments.
+./update_dependency.sh
+
+# "chatqna" here refers to the directory name that contains the Helm 
+# chart for the ChatQnA application 
+helm dependency update chatqna
+```
+
+Set the necessary environment variables to setup the use case
+```bash
+export MODELDIR="/mnt/opea-models"  #export MODELDIR="null" if you don't want to cache the model.  
```

Comment:
This caused an error:

```
Events:
  Type     Reason       Age                 From               Message
  ----     ------       ----                ----               -------
  Normal   Scheduled    10m                 default-scheduler  Successfully assigned default/chatqna-tei-645548d7f-7crjj to minikube
  Warning  FailedMount  17s (x13 over 10m)  kubelet            MountVolume.SetUp failed for volume "model-volume" : hostPath type check failed: /mnt/opea-models is not a directory
```

Using this command `export MODELDIR="null"` solved the issue.

Reply:
I would not recommend to set global.modelUseHostPath=${MODELDIR} in `helm install` in multi-node environment. This options is meant to share the pre downloaded model data files in a single node environment for a quick setup to test. To use this in a multiple nodes environment , you need to go to every K8S worker node to make sure there is a `${MODELDIR}` directory exists and writable.  Just don't set global.modelUseHostPath and leave it as the default value(empty string) which will download the models without sharing.

Another option to share the model files is to set global.modelUsePVC to use K8S persistent volume to share the model data files. In order to use that, users need to do some preparation work out-of-band:

1. Setup the K8S storage class, persistent volumes.
2. Create a K8S persistent volume claim which can be shared by multi pods.
3. pass the created k8s persistent volume claim to global.modelUsePVC

Reply:
I would not recommend setting it empty either, because not sharing the data will much more easily cause node to run out disk space, meaning that k8s evicts pods from it, and things in general break.

Reply:
@eero-t @lianhao 
Could we leave it empty and point to the following link to set up pvc?

https://github.com/opea-project/GenAIInfra/blob/56674fcc5ad76cdcf9f17377cef4f5ff8e3b90a4/helm-charts/README.md#using-persistent-volume:~:text=their%20usage%20metrics.-,Using%20Persistent%20Volume,-It%27s%20common%20to

Reply:
Linked instruction is fairly OK, but it assumes user knows something about setting up NFS:
* security implication of non-authenticated access
* need to replace IP address with the correct one for the used NFS server

Some knowledge of how PVCs work may also be good in case one needs to debug why pods are in `Pending` state (is the problem in NFS settings, PVC not in same namespace with pod using it, there being too many PVCs for given PV etc).